test/dotnet-add-proj.Tests/GivenDotnetAddProj.cs

```diff
@@ -0,0 +1,236 @@
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "AppAndLib.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\"");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("already contains project");
+            cmd.StdErr.Should().BeEmpty();
+
+            var contentAfter = File.ReadAllText(slnFullPath);
+            contentAfter.Should().BeEquivalentTo(contentBefore);
+        }
+
+        [Fact]
+        public void WhenPassedMultipleProjectsAndOneOfthemDoesNotExistItCancelsWholeOperation()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "App.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute($"\"{setup.GetProjectFullPath("Lib")}\" \"IDoNotExist.csproj\"");
+            cmd.Should().Fail();
+            cmd.StdErr.Should().Contain("does not exist");
+            cmd.StdErr.Should().NotMatchRegex("(.*does not exist.*){2,}");
+
+            var contentAfter = File.ReadAllText(slnFullPath);
+            contentAfter.Should().BeEquivalentTo(contentBefore);
+        }
+
+        [Fact]
+        public void WhenPassedProjectDoesNotExistAndForceSwitchIsPassedItAddsIt()
+        {
+            var setup = CreateTestSetup(TestGroup, ProjectName);
+            string slnDir = Path.Combine(setup.TestRoot, "MoreThanOne");
+            string slnName = "App.sln";
+            string slnFullPath = Path.Combine(slnDir, slnName);
+
+            var contentBefore = File.ReadAllText(slnFullPath);
+            var cmd = new AddProjCommand()
+                .WithWorkingDirectory(slnDir)
+                .WithSolution(slnName)
+                .Execute("--force \"IDoNotExist.csproj\"");
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("added to the solution");
+            cmd.StdErr.Should().BeEmpty();
+        }
+
+        private void VerifyProjectInSolution(SlnFile slnFile, string projectName, string projectGuid)
+        {
+            var projects = slnFile.Projects
+                .Where((p) => p.Name == projectName)
+                .ToList();
+
+            projects.Count.Should().Be(1);
+            if (!string.IsNullOrEmpty(projectGuid))
+            {
+                projects[0].Id.Should().Be(projectGuid);
+            }
+            projects[0].TypeGuid.Should().Be(SlnFile.CSharpProjectTypeGuid);
```

Comment:
I think the current F# project type guid is fine; we're in a bit of a funky situation for a while since we're still on our own project system, but will be moving to the Roslyn project system in the next year (at least that's the plan...).